api/v1beta1/azurecluster_default.go

```diff
@@ -128,6 +130,12 @@ func (c *AzureCluster) setSubnetDefaults() {
 			if subnet.NatGateway.NatGatewayIP.Name == "" {
 				subnet.NatGateway.NatGatewayIP.Name = generateNatGatewayIPName(c.ObjectMeta.Name, subnet.Name)
 			}
+		} else {
+			if !subnet.IsIPv6Enabled() {
+				// Keep LB support for IPv6 clusters, otherwise use NAT Gateway for outbound traffic
```

Comment:
would be good to specify why in the comment, i.e. `NAT gateway supports the use of IPv4 public IP addresses for outbound connectivity `